### ISSUE TITLE
adapt stellarium to non-android/non-aosp-based mobile devices? postmarket, ubuntoutuch, manjaro, maemo...

### ISSUE THREAD

**Issue by lm2lm2** [Stellarium/stellarium#3850]
hello

in order to definitely leave android/lineage/aosp operating system, to postmarketos (or others kind of mobian/droidian/maemoleste/ubuntutouch/manjaro), would it be possible to adapt stellarium on the pure linux mobile OS?

thank you!

**Comment by gzotti** [comment]
It is certainly "possible" to do that. But seriously the 3 of us won't do it. Be the one, do it and get famous!

**Comment by alex-w** [comment]
Suggestion: 10 years ago option GUI_MODE was removed as outdated solution, so, anyone can add separate gui for small-screen devices and reintroduce this mode again

**Comment by gzotti** [comment]
Hm? We have `GUI_MODE=Standard|None`. Was there a third option? A new option without large-screen Qt widgets (e.g. QML) could be added, but is far from a weekend enjoyment. This is what I mean. If @lm2lm2 so much wants it, here is the starting point.

**Comment by lm2lm2** [comment]
> Hm? We have `GUI_MODE=Standard|None`. Was there a third option? A new option without large-screen Qt widgets (e.g. QML) could be added, but is far from a weekend enjoyment. This is what I mean. If @lm2lm2 so much wants it, here is the starting point.

my thought would be : seeing the very great and perfect result of stellarium on android and iphone, does a such projects could aim the same interface, the same features, within a postmarketos (or other non android based) phone?

(in addition please notice i never coded...)

**Comment by 10110111** [comment]
> seeing the very great and perfect result of stellarium on android and iphone

If you mean Stellarium Mobile, it's a separate project, specifically designed to be used on mobile devices and sharing no code with Stellarium.

> if there is only "standard" and "none", how does it works for ios/android?

Desktop Stellarium doesn't work on iOS/Android AFAIK.

**Comment by lm2lm2** [comment]
> > seeing the very great and perfect result of stellarium on android and iphone
> 
> If you mean Stellarium Mobile, it's a separate project, specifically designed to be used on mobile devices and sharing no code with Stellarium.
> 
> > if there is only "standard" and "none", how does it works for ios/android?
> 
> Desktop Stellarium doesn't work on iOS/Android AFAIK.

but we can well concur that stellarium mobile is not 100% independent of the opensource stellarium app, not just a "name borrowed", but it's well a part of the desktop project, as code plus space/data information used, to adapt it on mobile?

**Comment by alex-w** [comment]
Old Stellarium Mobile is based on version 0.13 of desktop edition, but the modern Stellarium Mobile Plus is a full new project who sharing ideas and some look, but not the code (C language for Mobile+ and C++ language for Mobile)

**Comment by gzotti** [comment]
The current smartphone apps called "Stellarium mobile" for iOS and Android are not derived from the desktop program. Many years ago there was a smartphone version of Stellarium that was forked away from desktop Stellarium. It has developed too far away to remain source compatible. Some other people have forked and compiled desktop Stellarium for Android. These seem to work on ~10" tablets, but not on 6" phone screens. Their changes never went back here, so are not part of this project. Nobody, including any of the core developers, has ever provided a post-1.0 version of Desktop Stellarium running on Android devices with small screens.

**Comment by alex-w** [comment]
Please check https://github.com/ultrapre/Stellarium-mobile

**Comment by lm2lm2** [comment]
> The current smartphone apps called "Stellarium mobile" for iOS and Android are not derived from the desktop program. Many years ago there was a smartphone version of Stellarium that was forked away from desktop Stellarium. It has developed too far away to remain source compatible. Some other people have forked and compiled desktop Stellarium for Android. These seem to work on ~10" tablets, but not on 6" phone screens. Their changes never went back here, so are not part of this project. Nobody, including any of the core developers, has ever provided a post-1.0 version of Desktop Stellarium running on Android devices with small screens.

that means its quite definitely dead, to adapt either :
-stellarium desktop
-stellarium mobile project, android/iphone

..to any linux-phone, such as postmarketos, manjrao, mobian, maemo leste, ubuntu, sailfish..

**Comment by lm2lm2** [comment]
> Please check https://github.com/ultrapre/Stellarium-mobile

>Stellarium MobileStellarium Mobile is a [fork](https://en.wikipedia.org/wiki/Fork_(software_development)) of Stellarium, developed by some of the Stellarium team members. It currently targets mobile devices running [Symbian](https://en.wikipedia.org/wiki/Symbian), [Maemo](https://en.wikipedia.org/wiki/Maemo), [Android](https://en.wikipedia.org/wiki/Android_(operating_system)), and [iOS](https://en.wikipedia.org/wiki/IOS). Some of the mobile optimisations have been integrated into the mainline Stellarium product.
https://en.wikipedia.org/wiki/Stellarium_(software)

no real/easy clue to see a real independence between those two softwares

**Comment by alex-w** [comment]
> no real/easy clue to see a real independence between those two softwares

I share the link to fork an old mobile edition, who is based on desktop edition of Stellarium

**Comment by gzotti** [comment]
This is a fork of the version mentioned in
>Many years ago there was a smartphone version of Stellarium that was forked away from desktop Stellarium. It has developed too far away to remain source compatible.

**Comment by 10110111** [comment]
> no real/easy clue to see a real independence between those two softwares

Right, this one, despite being named "Stellarium-mobile", is not the app that you see in the app stores.

The actual independent project seen in the app stores is based on [this one](https://github.com/Stellarium/stellarium-web-engine), has seriously evolved and is no longer open-source.

**Comment by freebrowser1** [comment]
The Android version of Stellarium is very limited.
I have installed Andronix, a prooted Linux on Android with a complete desktop available via VNC.
Running Stellarium 24.2 (just by installing it via the Ubuntu package manager) is flawlessly, so I can run full fledged Stellarium (and other apps, like Firefox) on my smartphone. It is accessible via a VNC viewer on the smartphone itself (screen might be too small ?) or a VNC viewer on a desktop computer.
There are more Linux deploys on Android.

**Comment by gzotti** [comment]
There are Andro* power users who attach a desktop monitor or run VNC. And there are too few developers who add or improve functional features to the desktop-designated software in front of a 30" or larger desktop screen and just have no time to care for *droid and 12 flavours of free competitors. If you know what to do, fine. If you can describe what you do for others to repeat, much better. If you fork the repo, add a GUI that fulfills the needs of 6" Android screens (which you know better than devs on 30" screens whose smartphones die before they have learned all features), compiles for Android on at least one of the 3 supported desktop dev platforms (Windows, macOS, Ubuntu Linux, all with Qt6.5 or later), spits out an APK for self installation for e.g. Android8+ and whatever mobile systems you feel the need to support, clearly describes the configuration steps for total mobile dev noobs and does not break the sources for the desktop mainline, we could even include it in the official repo. But don't expect us to do this.

**Comment by peterjc** [comment]
See also https://github.com/sailfishos-chum/Stellarium-sailfishos for SailfishOS (as used on Jolla's 2026 Maemo based Linux phone) based on https://github.com/chengxinlun/Stellarium-android (based on Stellarium-Mobile) and assets from https://noctua-software.com/stellarium-mobile 1.29 version.

I presume that's the old "Stellarium Mobile" which @alex-w said above was based on version 0.13 of desktop edition (unrelated to the more modern "Stellarium Mobile Plus" rewrite in C).

Does that mean Stellarium-sailfishos is essentially a distant descendent of https://github.com/Stellarium/stellarium ?
